### PR TITLE
dhcp_server: Add DHCP Options + Enhanced Docs

### PR DESCRIPTION
**Pull request by rkmfl**
#### Context and Rationale

The `addon_core_dhcp_server` utilizes the standard ISC DHCP server which supports the full DHCP standard.

The current Home Assistant YAML only accepts three named parameters per host (ip, mac, name).

This limits the ability to specify other DHCP options such as *Option 66 (TFTP Server)*, which is essential for scenarios like network booting or flashing firmware to bricked devices.

DHCP Options have a variety of home usage, including specifying alternate domains or name servers for certain hosts.

Because this feature is already implemented, it's simply a matter of adding validation to the YAML enumerator that translates to the `dhcpd.conf`.

In addition to the enhancement, I've thoroughly edited the documentation for the add-on.

#### Proposed Changes

1. **YAML Parsing:**
   - Add validation for DHCP Options available in the dhcpd.conf

2. **Documentation**
   - Add explanation and examples. Proof/edit existing documentation for clarity/accuracy/readability.

#### Benefits

- **Enhanced Flexibility:** Users can define any standard DHCP options, increasing the flexibility and capability of the DHCP server.

- **User Empowerment:** Advanced users can leverage the full feature set of the ISC DHCP server without waiting for specific parameters to be implemented in Home Assistant.

#### Example YAML Structure:

```yaml
hosts:
  - name: MyRouter
    mac: aa:bb:cc:dd:ee:ff
    ip: 192.168.1.1
    6: 94.140.14.15  # Option 6 w/Family-Friendly DNS Server (AdGuard)
    66: my.tftp-server.internal
```

## Summary by CodeRabbit

- **Documentation**
  - Improved clarity and consistency in the DHCP Server add-on installation instructions and configuration details.

- **New Features**
  - Introduced a script that generates a DHCP server configuration based on user-defined parameters and starts the DHCP server.

- **Refactor**
  - Enhanced the `run.sh` script to dynamically create DHCP hosts and handle various DHCP options based on key types.

### CODE REVIEW

**Comment by rkmfl** [comment]
> It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-

Thank you. Signature completed.

File: dhcp_server/DOCS.md [hunk 1]
```diff
@@ -1,34 +1,34 @@
-# Home Assistant Add-on: DHCP server
+# Home Assistant Add-on: DHCP Server
```
Comment (agners) [inline dhcp_server/DOCS.md:1]: We generally use [sentence style capitalization](https://learn.microsoft.com/en-us/style-guide/text-formatting/using-type/use-sentence-style-capitalization), also [for titles](https://learn.microsoft.com/en-us/style-guide/text-formatting/formatting-titles), as per Microsoft writing style guideline.

File: dhcp_server/DOCS.md [hunk 2]
```diff
@@ -1,34 +1,34 @@
...
+3. Click the "INSTALL" button.
+4. Modify the configuration as needed.
 
-## How to use
+## How to Use
 
-1. Set the `domain` option, e.g., `mynetwork.local`.
-2. Save the add-on configuration by clicking the "SAVE" button.
+1. Set the `domain` option, e.g., `mynetwork.internal`.
+2. Save the configuration by clicking the "SAVE" button.
```
[the remark this answers is not in the document]
Comment (agners) [reply dhcp_server/DOCS.md:15]: The current guide is fine.

File: dhcp_server/DOCS.md [hunk 3]
```diff
@@ -38,99 +38,126 @@ networks:
     range_end: 192.168.1.200
     broadcast: 192.168.1.255
     gateway: 192.168.1.1
-    interface: end0
+    interface: eth0
```
Comment (agners) [inline dhcp_server/DOCS.md:41]: Almost none of the OS installations by default use `eth0` these days. Someone even recently asked to change that https://github.com/home-assistant/addons/pull/3638 :sweat_smile: 

So let's leave it as `end0` now.

File: dhcp_server/DOCS.md [hunk 4]
```diff
@@ -38,99 +38,126 @@ networks:
...
 
-The MAC address of the client device.
+The MAC address of the host (client) device.
 
 #### Option: `hosts.ip`
 
-The IP address you want the DHCP server to assign.
+The IP address assigned to the host (client) device.
+
+#### Option: Number between 1 - 254
```
Comment (agners) [inline dhcp_server/DOCS.md:150]: I don't think the Supervisor/Frontend allows to set these keys in the add-on config without adding them to the schema in config.yaml no?

What we could do here as a work around is add a config option `extra_dhcp_options` which is a list of itmes with a `key` and `value` property.

**Review by agners — CHANGES_REQUESTED** [review]
Thanks for your contribution!

You also need to bump the add-on version in `config.yaml` and add an entry to the CHANGELOG.md file.

**Pull request closed without merging**